dev/source/docs/setting-up-sitl-on-macosx.rst

```diff
@@ -0,0 +1,137 @@
+hardware. It is a build of the autopilot code using an ordinary C++
+compiler, giving you a native executable that allows you to test the
+behaviour of the code without hardware.
+
+SITL runs natively on Linux, Windows and Mac OSX.
+
+.. image:: ../images/SITL_macosx.png
+    :target: ../_images/SITL_macosx.png
+
+
+Install steps
+=============
+
+This guide assumes that a build environment is already setup for Ardupilot. If that is not the case please follow the available guides to setup the build environment for Mac OSX. The following step provide a condensed versions of the steps to take.
+
+
+Prerequisites
+-------------
+
+The following tools need to be installed to complete the installation. Between brackets the version used by the author at time of writing.
+
+ * Xcode (12.4)
+ * Xcode commandline tools (12.4)
+ * Homebrew (3.0.4-52-gbc37074)
+
+Preparing Python
+----------------
+
+Install the required python 3 version using homebrew and activate it. If you are familiar with pyenv or virtualenv it might be better to use one of those to setup the environment. This limits potential problems with other tools that require specific versions of python. How to do that is beyond the scope of this document.
+
+The following steps install python version 3.9.x for system wide usage.
+
+::
+
+	brew install python@3.9
+  export PATH="$(brew --prefix)/opt/python/libexec/bin":$PATH
+	
+	pip install wheel
+	pip install future pymavlink mavproxy
+
+Optional: Add the following to the either .bashrc or .zshrc to enable this version of python be default
+
+::
+
+  export PATH="$(brew --prefix)/opt/python/libexec/bin":$PATH
+
+
+Optional: Install sfml (used by sitl for audio and osd)
+
+::
+
+  brew install sfml
+
+Note that not all sfml features work on Mac OSX currently. The following error can appear `Cannot create a window from a worker thread. (OS X limitation)`, this is a result of Mac OSX specifics that haven't been catered for in the sitl code yet.
+
+Start SITL simulator
+--------------------
+
+.. warning::
+
+	Due to ideosyncracies with Python and using screens there are limitations when Python needs to use the screen to display a graphical UI. Python 3.9 solves part of the problem by no longer requiring a separate pythonw binary. However the requirement to run on the main display still stands. When observing the error "This program needs access to the screen. Please run with a Framework build of python, and only when you are logged in on the main display of your Mac.", disconnect external displays during startup. When SITL is started the windows can be moved around as normal.
+
+To start the simulator first change directory to the vehicle directory.
+For example, for the multicopter code change to **ardupilot/ArduCopter**:
+
+::
+
+   cd ardupilot/ArduCopter
+
+The sim_vehicle script checks several environment variables to determine the OS it's running on. For Mac OSX it
+requires that DISPLAY is set.
+
+::
+
+  export DISPLAY=:0.0
```

Comment:
Don't you need the macOS X environment installed for this to work?

Reply:
I've not needed to set the DISPLAY when running SITL (no need for an X server that I'm aware of?)